gravitee-am-reporter/gravitee-am-reporter-kafka/src/main/java/io/gravitee/am/reporter/kafka/audit/KafkaAuditReporter.java

```diff
@@ -110,7 +112,17 @@ protected void doStart() {
     protected void doStop() throws Exception {
         super.doStop();
         if (this.producer != null) {
-            this.producer.close();
+            // Need to do unwrap producer because of bug in kafka vertx library: https://github.com/vert-x3/vertx-kafka-client/issues/271
+            try (Producer<String, AuditMessageValueDto> producer = this.producer.unwrap()) {
```

Comment:
**question**: is it usefull to use a try resources here as you are calling the close method in the try block ?

Reply:
unwrap is autoclosable. Thats why it is in try-with-resource.

Reply:
yes I know, but my point is the close provided by the unwrap is closing the same resource (KafkaWriteStream<K, V> stream) as the one you are closing in the try statement. So you are closing two times the same thing.

If there is no concurrent issue for doing this, then it is OK.

Reply:
I got it. Removed try block to not call close twice.